### ISSUE TITLE
Error: invalid argument for "-e, --env" flag: parse error on line X, column Y: bare " in non-quoted-field

### ISSUE THREAD

**Issue by cevich** [containers/libpod#2810]
/kind bug

**Description**

Attempting to pass env. vars. containing quotes through ``--env`` returns error.

**Steps to reproduce the issue:**

1. ``$ BAD="=\"A='B=\"C='D=E=F=G"``
2.  ``$ echo $BAD``
3. ``$ echo "$BAD"``
2. ``$ sudo podman run -it --rm -e "BAD=$BAD" fedora bash``

**Describe the results you received:**

```
echo $BAD
="A='B="C='D=E=F=G

$ echo "$BAD"
="A='B="C='D=E=F=G

$ sudo podman run -it --rm -e "BAD=$BAD" fedora bash
Error: invalid argument "BAD==\"A='B=\"C='D=E=F=G" for "-e, --env" flag: parse error on line 1, column 5: bare " in non-quoted-field
```

**Describe the results you expected:**

```
$ sudo podman run -it --rm -e "BAD=$BAD" fedora bash
[root@4a43b0055b2f /]# echo $BAD
="A=\'B="C=\'D=E=F=G
[root@4a43b0055b2f /]# echo "$BAD"
="A=\'B="C=\'D=E=F=G
```

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
Version:            1.1.2
RemoteAPI Version:  1
Go Version:         go1.11.5
Git Commit:         a95a49d3038462d033f84ac314ec8a3064a99cff
Built:              Tue Mar  5 13:10:31 2019
OS/Arch:            linux/amd64
```

**Output of `podman info --debug`:**

```
debug:
  compiler: gc
  git commit: a95a49d3038462d033f84ac314ec8a3064a99cff
  go version: go1.11.5
  podman version: 1.1.2
host:
  BuildahVersion: 1.7.1
  Conmon:
    package: podman-1.1.2-1.git0ad9b6b.fc29.x86_64
    path: /usr/libexec/podman/conmon
    version: 'conmon version 1.12.0-dev, commit: a95a49d3038462d033f84ac314ec8a3064a99cff'
  Distribution:
    distribution: fedora
    version: "29"
  MemFree: 1238302720
  MemTotal: 12437794816
  OCIRuntime:
    package: runc-1.0.0-83.dev.gitf56b4cb.fc29.x86_64
    path: /usr/bin/runc
    version: |-
      runc version 1.0.0-rc6+dev
      commit: cb19e6a579f2a7da040525d725cfb871b4275148
      spec: 1.0.1-dev
  SwapFree: 831557632
  SwapTotal: 1073737728
  arch: amd64
  cpus: 4
  hostname: ...cut...
  kernel: 4.20.16-200.fc29.x86_64
  os: linux
  rootless: true
  uptime: 266h 42m 40s (Approximately 11.08 days)
insecure registries:
  registries:
  - localhost
registries:
  registries:
  - docker.io
  - registry.fedoraproject.org
store:
  ConfigFile: /home/cevich/.config/containers/storage.conf
  ContainerStore:
    number: 0
  GraphDriverName: overlay
  GraphOptions:
  - overlay.mount_program=/usr/bin/fuse-overlayfs
  GraphRoot: /home/cevich/.local/share/containers/storage
  GraphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  ImageStore:
    number: 0
  RunRoot: ...cut...
  VolumePath: /home/cevich/.local/share/containers/storage/volumes
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

A similar string without the embedded quotes ``BAD="=A=B=C=D=E=F=G"`` works fine.

**Comment by mheon** [comment]
This one ought to be fixed on master

**Comment by cevich** [comment]
Confirmed: Fixed in 1.2.0
